Popup/tests/Popup-specs.js

```diff
@@ -1,236 +1,309 @@
 		});
 	});
 
-	describe('with position changes dynamically - [GT-28270]', function () {
-		it('should not have top class when position change from top to any other position', () => {
+	describe('with position changes dynamically', function () {
+		test('should not have top class when position change from top to any other position', () => {
 			const firstPosition = 'top';
-			const popup = shallow(
-				<PopupBase open position={firstPosition}><div>popup</div></PopupBase>
+			const {getByRole, rerender} = render(
+				<FloatingLayerController>
+					<Popup open position={firstPosition}><div>popup</div></Popup>
+				</FloatingLayerController>
 			);
 
-			expect(popup.find(`.${css.popup}`).prop('className').split(' ')).toContain(firstPosition);
-			expect(popup.prop('className').split(' ')).toContain(firstPosition);
-			expect(popup.find(`.${css.popup}`).prop('className').split(' ')).not.toContain('fullscreen');
-			expect(popup.prop('className').split(' ')).not.toContain('fullscreen');
-			expect(popup.find(`.${css.popup}`).prop('className').split(' ')).not.toContain('center');
-			expect(popup.prop('className').split(' ')).not.toContain('center');
-			expect(popup.find(`.${css.popup}`).prop('className').split(' ')).not.toContain('bottom');
-			expect(popup.prop('className').split(' ')).not.toContain('bottom');
-			expect(popup.find(`.${css.popup}`).prop('className').split(' ')).not.toContain('right');
-			expect(popup.prop('className').split(' ')).not.toContain('right');
-			expect(popup.find(`.${css.popup}`).prop('className').split(' ')).not.toContain('left');
-			expect(popup.prop('className').split(' ')).not.toContain('left');
-
-			popup.setProps({position: 'fullscreen'});
+			const popup = getByRole('alert');
+			const transitionContainer = getByRole('alert').parentElement.parentElement;
+
+			expect(popup.className).toContain(firstPosition);
+			expect(transitionContainer.className).toContain(firstPosition);
+			expect(popup.className).not.toContain('fullscreen');
+			expect(transitionContainer.className).not.toContain('fullscreen');
+			expect(popup.className).not.toContain('center');
+			expect(transitionContainer.className).not.toContain('center');
+			expect(popup.className).not.toContain('bottom');
+			expect(transitionContainer.className).not.toContain('bottom');
+			expect(popup.className).not.toContain('right');
+			expect(transitionContainer.className).not.toContain('right');
+			expect(popup.className).not.toContain('left');
+			expect(transitionContainer.className).not.toContain('left');
+
+			rerender(
+				<FloatingLayerController>
+					<Popup open position="fullscreen"><div>popup</div></Popup>
+				</FloatingLayerController>
+			);
 
-			expect(popup.find(`.${css.popup}`).prop('className').split(' ')).not.toContain(firstPosition);
-			expect(popup.prop('className').split(' ')).not.toContain(firstPosition);
+			expect(popup.className).not.toContain(firstPosition);
```

Comment:
I'm not sure the original author's intention but shouldn't we check "fullscreen" class?

Reply:
I'm not sure either, but the test description is "should not have top class when position change from top to any other position", so I think we are just checking it's not "top".